client/common/cmdline.c

```diff
@@ -830,15 +831,26 @@ static int freerdp_client_command_line_post_filter(void* context, COMMAND_LINE_A
 	}
 	CommandLineSwitchCase(arg, "smartcard")
 	{
-		char** p;
-		int count;
-
-		p = freerdp_command_line_parse_comma_separated_values_offset(arg->Value, &count);
-		p[0] = "smartcard";
-
-		status = freerdp_client_add_device_channel(settings, count, p);
-
-		free(p);
+		if (arg->Flags & COMMAND_LINE_VALUE_PRESENT)
+		{
+			char** p;
+			int count;
+			
+			p = freerdp_command_line_parse_comma_separated_values_offset(arg->Value, &count);
+			p[0] = "smartcard";
+			free(p);
```

Comment:
@HenryJacques This is definitely broken, you do not load the channel and free the arguments...